invenio_records_resources/serializers/xml.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+# Copyright (C) 2020 Northwestern University.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Invenio Resources basic XML serialization module."""
+
+import xmltodict
+from flask_resources.serializers import SerializerMixin
+
+from .json import RecordJSONSerializer
+
+
+class RecordXMLSerializer(SerializerMixin):
+    """Record JSON serializer implementation."""
+
+    def serialize_object(self, obj, response_ctx=None, *args, **kwargs):
+        """Dump the object into an XML string."""
+        if obj:  # e.g. delete op has no return body
+            json_serializer = RecordJSONSerializer()
+            processed_record = json_serializer._process_record(obj)
+            return xmltodict.unparse({"root": processed_record})
```

Comment:
Moderate/Major: I understand this is for "demo" purposes. However, dumping json here makes the whole object end up in a single key - `root`. IMHO, each key should be its own XML one.

```python
>>> a = {"test": 1, "another": 2}
>>> xmltodict.unparse({"root":a})
'<?xml version="1.0" encoding="utf-8"?>\n<root><test>1</test><another>2</another></root>'
>>> xmltodict.unparse({"root":json.dumps(a)})
'<?xml version="1.0" encoding="utf-8"?>\n<root>{"test": 1, "another": 2}</root>'
```

```suggestion
            return xmltodict.unparse({"root": obj})
```

I think this is not noticed because the integration tests do not check the contents of the `metadata` field, which is where the output of this function ends up.

extra karma: removes the dependency form `JSONSerializer` 😇

Reply:
Oh, this is subtle :neckbeard: :

```python
xmltodict.unparse({"root": json.dumps(a)})
'<?xml version="1.0" encoding="utf-8"?>\n<root>{"test": 1, "another": 2}</root>'
```

is because `json.dumps(a)` returns a string. However, `json_serializer._process_record(obj)` returns a `dict` and that makes the whole difference. So we are good here! (the tests would have broken because we are iterating on the root and getting xml `Element` objects that respond to the `tag()` message. `dict` keys don't. ;) )

Reply:
Hmm... you are right. Sorry I missunderstood that. Still I wouldn't make one serializer dependent on the other. We can make an independent method out of `_process_record`? It does actually not relate to any `self`.

Reply:
I think extracting _process_record makes sense, but I would wait a little bit to determine where to extract it. I think the works we are both doing will help us see that. For now, I would say let's go with that.